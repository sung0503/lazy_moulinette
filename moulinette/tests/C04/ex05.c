#include <stdio.h>

int ft_atoi_base(char *str, char *base);

void test(char *str, char *base)
{
    int res = ft_atoi_base(str, base);
    printf("%d\n", res);
}

int main(void)
{
    test("4242", "");
    test("4242", "4");
    test("1000010010010", "01");
    test("4242", "012345678+");
    test("4242", "012345678-");
    test("4242", "012345678 ");
    test("4242", "012345678\t");
    test("4242", "012345678\n");
    test("4242", "012345678\v");
    test("4242", "012345678\f");
    printf("---\n");
    test("4242", "abcdefghij");
    printf("---\n");
    test(" \t\n\v\f+1234", "0123456789");
    test(" \t\n\v\f+-1234", "0123456789");
    test(" \t\n\v\f+--1234", "0123456789");
    test(" \t\n\v\f-+-1234", "0123456789");
    test(" \t\n\v\f-+--1234", "0123456789");
    test(" \t\n\v\f-+-1234a", "0123456789");
    test(" \t\n\v\f-+-123a4", "0123456789");
    test(" \t\n\v\f-+-12a34", "0123456789");
    test(" \t\n\v\f-+-1a234", "0123456789");
    test(" \t\n\v\f-1234-", "0123456789");
    test(" \t\n\v\f+1234-", "0123456789");
    test(" \t\n\v\f+- 1234-", "0123456789");
    printf("---\n");
    test(" \t\n\v\f-+-IOI", "OI");
    test(" \t\n\v\f-+-!@!", "@!");
    test(" \t\n\v\f-+-^%^", "%^");
    printf("---\n");
    test("4242", "0123456788");
    test("4242", "0123456784");
    return 0;
}
