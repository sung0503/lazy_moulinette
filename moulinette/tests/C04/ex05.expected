0
0
4242
0
0
0
0
0
0
0
---
0
---
1234
-1234
1234
1234
-1234
1234
123
12
1
-1234
1234
0
---
5
5
5
---
0
0
